test/asynchronous/unified_format.py

```diff
@@ -630,6 +630,9 @@ def process_error(self, exception, spec):
                 self.assertNotIsInstance(error, NotPrimaryError)
             elif isinstance(error, (InvalidOperation, ConfigurationError, EncryptionError)):
                 pass
+            # gridfs NoFile errors are considered client errors.
+            elif isinstance(error, NoFile):
```

Comment:
Suggest adding this to the previous elif: `InvalidOperation, ConfigurationError, EncryptionError, NoFile)):`